### PR TITLE
Move version info outside of HB

### PR DESCRIPTION
**Pull request by heroicNeZha**
#### What type of PR is this?
- [ ] bug
- [ ] feature
- [x] enhancement

#### What does this PR do?
1. Move the version info outside the heartbeat
2. Save the version info when establishing the connection to metad server.

#### Which issue(s)/PR(s) this PR relates to?
#3055 

#### Checklist：
- [ ] Documentation affected （Please add the label if documentation needs to be modified.)
- [x] Incompatible （If it is incompatible, please describe it and add corresponding label.）
- [ ] Need to cherry-pick （If need to cherry-pick to some branches, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

#### Release notes：
Please confirm whether to reflect in release notes and how to describe:
>                                                                 `

### CODE REVIEW

**Comment by darionyaphet** [comment]
why remove the version from heartbeat?

**Comment by heroicNeZha** [comment]
> why remove the version from heartbeat? to fix any problem?

it seems not necessary

File: src/meta/test/VerifyClientVersionTest.cpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
+/* Copyright (c) 2018 vesoft inc. All rights reserved.
```
Comment (Aiee) [inline src/meta/test/VerifyClientVersionTest.cpp:1]: 2021 vesoft

File: src/meta/test/VerifyClientVersionTest.cpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+#include "meta/processors/admin/VerifyClientVersionProcessor.h"
+#include "meta/processors/parts/ListHostsProcessor.h"
+#include "meta/test/TestUtils.h"
+
+namespace nebula {
+namespace meta {
+
+TEST(VerifyClientVersionTest, VersionTest) {
+  fs::TempDir rootPath("/tmp/VersionTest.XXXXXX");
+  std::unique_ptr<kvstore::KVStore> kv(MockCluster::initMetaKV(rootPath.path()));
```
Comment (Aiee) [inline src/meta/test/VerifyClientVersionTest.cpp:20]: Please add some failure cases.

File: src/meta/processors/admin/HBProcessor.cpp [hunk 3]
```diff
@@ -46,9 +46,6 @@ void HBProcessor::process(const cpp2::HBReq& req) {
   }
 
   HostInfo info(time::WallClock::fastNowInMilliSec(), req.get_role(), req.get_git_info_sha());
-  if (req.version_ref().has_value()) {
-    info.version_ = *req.version_ref();
```
Comment (yixinglu) [inline src/meta/processors/admin/HBProcessor.cpp:50]: Need you to delete the `version_` from `HostInfo` struct ?

Comment (heroicNeZha) [reply src/meta/processors/admin/HBProcessor.cpp:50]: I haven’t deleted it yet, but it’s no used. Do I need to delete it?

Comment (yixinglu) [reply src/meta/processors/admin/HBProcessor.cpp:50]: yeah

Comment (heroicNeZha) [reply src/meta/processors/admin/HBProcessor.cpp:50]: done

File: src/interface/meta.thrift [hunk 4]
```diff
@@ -1106,6 +1104,7 @@ struct VerifyClientVersionResp {
 
 struct VerifyClientVersionReq {
     1: required binary version = common.version;
+    3: common.HostAddr host;
```
Comment (liuyu85cn) [inline src/interface/meta.thrift:1107]: why 3? not 2

**Comment by darionyaphet** [comment]
Some people remember why there is a version number in their heartbeat? @yixinglu @Aiee @liuyu85cn

File: src/interface/meta.thrift (hunk 4 above)
Comment (heroicNeZha) [reply src/interface/meta.thrift]: my mistake

**Review by yixinglu — APPROVED** [review]

**Comment by yixinglu** [comment]
> Some people remember why there is a version number in their heartbeat? @yixinglu @Aiee @liuyu85cn

Introduced in [this commit](https://github.com/vesoft-inc/nebula/commit/db76bf260ab8f987bb740021f388ca08e6ec8a4b)

**Review by CPWstatic — APPROVED** [review]

**Comment by HarrisChu** [comment]
How to keep the same version between storaged and metad?
e.g.
v2.5.1 storaged sends heartbeat to v2.6.1 metad

**Comment by heroicNeZha** [comment]
> How to keep the same version between storaged and metad? e.g. v2.5.1 storaged sends heartbeat to v2.6.1 metad

by whitelist in metad:   if v2.5.1 and v2.6.1 in whitelist, then they can in same cluster

**Comment by darionyaphet** [comment]
> 

Cluster Manager is in progress.

**Pull request merged**